test/hotspot/jtreg/runtime/cds/appcds/NestHostOldInf.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+/*
+ * @test
+ * @bug 8285914
+ * @summary A lambda proxy class should not be archived if its nest host implements an
+ *          old (with major version < JDK_6 (50) interface which cannot be verified during dump time).
+ * @requires vm.cds
+ * @library /test/lib
+ * @compile test-classes/OldInf.jasm
+ * @compile test-classes/ChildOldInf.java
+ * @compile test-classes/NestHostOldInfApp.java
```

Comment:
Can these not be combined into a single @compile - or is there some ordering dependency?

Reply:
Fixed.